docs/discussion.md

```diff
@@ -0,0 +1,197 @@
+         vertex_weight $`*=`$ (coord $\_{dim}\$ == 0? $1-\mu_{dim}$: $\mu_{dim}$);   
+    f[x] += f[x<sub>ij...k</sub>] $`*`$ vertex_weight;  
+return f(x);
+## 1-D Cubic
+Assume $f$ is a degree-3 polynomial:   
+     $\ f(x)=a\cdot x^3+b\cdot x^2+c\cdot x+d \$
+
+
+We need additional information, so let's use the slopes at $f(x_0)$ and $f(x_1)$: $f'(x_0)$ and $f'(x_1)$.
+
+Warning of notation change. I am shortening $f(x_i)$ to $f(i)$ at this point.
+
+
+<p align="center">
+    <img width="70%" src="./figs/fig_1Dcubic.png"> 
+</p>
+
+## 1-D Cubic: deriving coefficients
+The polynomial can be written:   
+     $f(\mu)=a\cdot \mu ^3+b\cdot \mu ^2+c\cdot \mu+d$
+
+Note that   
+     $f'=\dfrac{df}{dx}=\dfrac{d\mu}{dx} \cdot \dfrac{df}{d\mu}=\dfrac{1}{\delta_x}\cdot\dfrac{df}{d\mu}$:   
+
+Find the derivative:   
+     $f'(\mu)=(3a\cdot \mu^2+2b\cdot \mu+c)/\delta_x$   
+where $\delta_x=x_1-x_0$.
+
+Evaluate at 0 and 1:   
+     $f(0)=d$   
+     $f(1)=a+b+c+d$   
+     $f'(0)=c/\delta_x$   
+     $f'(1)=(3a+2b+c)/\delta_x$   
+
+Solve for coefficients [^1]:   
+     $a=2f(0)-2f(1)+\delta_x\cdot f'(0)+\delta_x\cdot f'(1)$   
+     $b=-3f(0)+3f(1)-2\delta_x\cdot f'(0)-\delta_x\cdot f'(1)$   
+     $c=\delta_x\cdot f'(0)$   
+     $d=f(0)$
+
+Substitute into $f(\mu)$:
+```math
+\begin{align*}
+f(\mu)=&(2\mu^3-3\mu^2+1)\cdot f(0)\\
++&(-2\mu^3+3\mu^2)\cdot f(1)\\
++&(\mu^3-2\mu^2+\mu)\cdot \delta_x\cdot  f'(0)\\
++&(\mu^3-\mu^2)\cdot \delta_x\cdot f'(1)\\
+\end{align*}
+```
+## 1-D Cubic
+We know $f(0)$ and $f(1)$. Those are our known grid values. But what about the derivatives?
+
+In Catmull-Rom splines, we use the slope of the line between the previous and next points on the axis as the derivative [^2]:   
+     $f'(0)=(f(1)-f(-1))/(x_1-x_{-1})$   
+     $f'(1)=(f(2)-f(0))/(x_2-x_0)$  
+
+Define:   
+     $c_{0}=2\mu^3-3\mu^2+1, \quad  c_{1}=-2\mu^3+3\mu^2$   
+     $d_{0}=\mu^3-2\mu^2+\mu, \quad d_{1}=\mu^3-\mu^2$  
+     $s_0=\delta_x/(x_1-x_{-1}), \quad s_1=\delta_x/(x_2-x_0)$
+
+Rewrite the polynomial:   
+ $f(\mu)=c_{0}\cdot f(0)+c_{1}\cdot f(1)+d_{0}\cdot s_0\cdot (f(1)-f(-1))+d_{1}\cdot s_1\cdot (f(2)-f(0))$  
+
+Group the coefficients:   
+ $f(\mu)=-d_{0}\cdot s_0\cdot f(-1)+(c_{0} -d_{1}\cdot s_1)\cdot f(0)+(c_{1}+d_{0}\cdot s_0)\cdot f(1)+d_{1}\cdot s_1\cdot f(2)$  
+
+## 1-D Cubic, Pseudocode
+ 
+f[x]=0;   
+weight = 1.0;   
+for i = -1 to 2:   
+    switch i:   
+        case -1: weight *= -d[0] $`*`$ s[0];     
+        case 0: weight *= c[0] - d[1] $`*`$ s[1];  
+        case 1: weight *= c[1] + d[0] $`*`$ s[0];   
+        case 2: weight *= d[1] $`*`$ s[1];  
+    f[x] += f[i] $\*\$ weight;  
+return f[x];
+
+Terms $c_{0}$ and $c_{1}$ are referred to programmatically as *interpolation coefficients*.
+
+The terms $d_{0}\cdot s_0$ and  $d_{1}\cdot s_1$ are referred to as *cubic_slope_coefficients*.
+
+## 2-D Cubic, with pictures
+There are 4<sup>N</sup> points to weigh. Number of calculations grows faster than that.
```

Comment:
A few sentences seem to have gotten fragmented; these just need to read more fluidly.